dashboard/api.py

```diff
@@ -207,27 +209,29 @@ def _add_run(run, mmtrack_, status):
     run_statuses.remove(run_status)
 
     if run_status.status == CourseRunStatus.NOT_ENROLLED:
-        next_run = course.get_next_run()
+        next_run = course.get_first_unexpired_run()
         if next_run is not None:
             _add_run(next_run, mmtrack, CourseStatus.OFFERED)
     elif run_status.status == CourseRunStatus.NOT_PASSED:
-        next_run = course.get_next_run()
+        _add_run(run_status.course_run, mmtrack, CourseRunStatus.NOT_PASSED)
+        next_run = course.get_first_unexpired_run()
         if next_run is not None:
             _add_run(next_run, mmtrack, CourseStatus.OFFERED)
-        if next_run is None or run_status.course_run.pk != next_run.pk:
-            _add_run(run_status.course_run, mmtrack, CourseStatus.NOT_PASSED)
+    elif run_status.status == CourseRunStatus.MISSED_DEADLINE:
+        _add_run(run_status.course_run, mmtrack, CourseRunStatus.MISSED_DEADLINE)
+        future_run = course.get_first_future_run()
```

Comment:
yeah, this touches on my [previous comment](https://github.com/mitodl/micromasters/pull/1389#issuecomment-254308767). the `MISSED_DEADLINE` status implies that the course run is current. `get_first_unexpired_run` would return the current course run, which we obviously don't want in this case. still, only getting future runs (ie: runs with start_date > now) is probably not correct here. one solution would be to add an optional parameter to `get_first_unexpired_run` that would exclude a specific course run/course run id. what do you think?